ndscan/to_txt.py

```diff
@@ -0,0 +1,89 @@
+"""Standalone tool to write scan data to text files"""
+
+import argparse
+import os
+import json
+import numpy as np
+
+from .show import load_h5
+from .results.tools import get_source_id
+from .utils import strip_suffix
+
+
+def get_argparser():
+    parser = argparse.ArgumentParser(
+        description="Extracts ndscan result data from ARTIQ HDF5 results file",
+        epilog=(
+            "Instead of a file name, just a run id or 'magic' source string can be "
+            "supplied, which is then resolved using oitg.results (e.g. 'alice_12345', "
+            "or just '12345' to infer the experiment name from the environment)."))
+    parser.add_argument("--prefix",
+                        default=None,
+                        type=str,
+                        help="Prefix of root in dataset tree (default: auto-detect)")
+    parser.add_argument("path",
+                        metavar="FILE",
+                        type=str,
+                        help="Path to HDF5 results file")
+    return parser
+
+
+def main():
+    args = get_argparser().parse_args()
+
+    path, datasets, prefixes, schema = load_h5(args)
+    if len(prefixes) > 1:
+        raise Exception("More than one ndscan prefix found. Please specify one. " +
+                        f"Prefixes: {prefixes}")
+    prefix = prefixes[0]
+
+    axes = json.loads(datasets[prefix + "axes"][()])
+    axes_names = [ax["param"]["fqn"].split(".")[-1] for ax in axes]
```

Comment:
Hmm, there could be two axes with the same parameter (e.g. a 2D scan of RSB π time vs carrier π time) – see ndscan.utils.shorten_to_unambiguous_suffixes`. For the same reason, the path (instead of fqn) would probably be more approrpriate, though the situation is a bit more complex for scans (overrides in general), as one can use the glob (`*`) matches. I guess we can live with this for now.